src/main.rs

```diff
@@ -207,7 +243,30 @@ fn main() -> Result<()> {
         }
     }
 
-    if !delete_file.exists() {
+    if cli.worktree_branch.is_some() {
+        if !delete_file.exists() {
+            println!(
+                "Working tree directory preserved at: {}",
+                tmp_dir.into_path().display()
+            );
+        } else {
+            let mut command = process::Command::new("git");
+            command.args(&[
+                "worktree",
+                "remove",
+                &tmp_dir
+                    .path()
+                    .file_name()
+                    .expect("Cannot get working tree name")
+                    .to_str()
+                    .expect("Cannot convert working tree name"),
```

Comment:
You don't need to convert `tmp_dir` to an str. You can pass it directly without conversion using `command.arg()`